tests/SideBySide.New/StoredProcedureTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Data;
+using System.Threading.Tasks;
+using MySql.Data.MySqlClient;
+using Xunit;
+
+namespace SideBySide
+{
+	public class StoredProcedureTests : IClassFixture<DatabaseFixture>, IDisposable
+	{
+		public StoredProcedureTests(DatabaseFixture database)
+		{
+			m_database = database;
+			m_database.Connection.Open();
+		}
+
+		public void Dispose()
+		{
+			m_database.Connection.Close();
+		}
+
+		[Fact]
+		public async Task StoredProcedureBasic()
+		{
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = @"DROP PROCEDURE IF EXISTS simple;
+CREATE PROCEDURE simple()
+BEGIN
+	SELECT 'simple';
+END";
+				await cmd.ExecuteNonQueryAsync();
+			}
+
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = "simple";
+				cmd.CommandType = CommandType.StoredProcedure;
+// MySql.Data makes the assumption that Stored Procedures with ParameterDirection.ReturnValue are functions
```

Comment:
I don't like the assumption `MySql.Data` makes about Stored Procedure vs. Function based off ParameterDirection.ReturnValue

Could we implement an extension of [System.Data.CommandType](https://msdn.microsoft.com/en-us/library/system.data.commandtype(v=vs.110).aspx) and include a `MySqlCommandType.Function` enum?